# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/layers/CanvasDrawEventRecorder.h
# Commit: b74ed8e9d95a
# Full Hash: b74ed8e9d95a8907add8dd764ccc89be1c7ab99d
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2023-12-13 09:27:26
# Description:
#   Bug 1869658 - Fix usage of CanvasBuffer on failure paths. r=gfx-reviewers,lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D196232
# ==============================================================================

diff -r 3f82f965831a -r b74ed8e9d95a gfx/layers/CanvasDrawEventRecorder.h
--- a/gfx/layers/CanvasDrawEventRecorder.h	Wed Dec 13 03:45:15 2023 +0000
+++ b/gfx/layers/CanvasDrawEventRecorder.h	Wed Dec 13 03:45:16 2023 +0000
@@ -147,7 +147,7 @@
                            aShmem->Size()),
           shmem(std::move(aShmem)) {}
 
-    size_t Capacity() { return shmem->Size(); }
+    size_t Capacity() { return shmem ? shmem->Size() : 0; }
   };
 
   struct RecycledBuffer {
